src/linker/Linker.Dataflow/ScannerExtensions.cs

```diff
@@ -39,6 +39,51 @@ public static HashSet<int> ComputeBranchTargets (this MethodBody methodBody)
 			}
 			return branchTargets;
 		}
+
+		public static HashSet<int> GetInitialBasicBlockInstructions (this MethodBody methodBody)
+		{
+			// Method identifies leading instructions in each basicBlock.
+			// An instruction defines a new basicBlock iff it is first instruction, jump target or instruction following jump target.
+			// This makes is different than ComputeBranchTargets, which returns only jump targets.
+			// Currently, the method does not support exception handling syntax.
+			var leaders = new HashSet<int> ();
+
+			foreach (Instruction operation in methodBody.Instructions) {
+
+				// First instruction is a leader
+				if (leaders.Count == 0)
+					leaders.Add (operation.Offset);
+
+				// Targets of control flow instructions are leaders
+				if (operation.OpCode.IsControlFlowInstruction ()) {
+					var jumpTargets = operation.GetJumpTargets ();
+
+					foreach (var target in jumpTargets) {
+						leaders.Add (target.Offset);
+					}
+
+					// Instructions following conditional or unconditional jumps are leaders
```

Comment:
Yes, the next instruction is not reachable from it. Currently, an unreachable basic block is created.

Reply:
OK - makes sense. There was a case where linker had problems which was basically when the method ended in unconditional jump (no instruction afterward), I assume this would lead to creation of an empty basic block, which is not a problem.